pkg/host/internal/sriov/sriov.go

```diff
@@ -467,76 +502,184 @@ func (s *sriov) ConfigSriovInterfaces(storeManager store.ManagerInterface,
-					if err != nil {
-						log.Log.Error(err, "SyncNodeState(): failed to save PF applied config to host")
-						return err
-					}
-
-					break
-				}
-				if err := s.ConfigSriovDevice(&iface, &ifaceStatus); err != nil {
-					log.Log.Error(err, "SyncNodeState(): fail to configure sriov interface. resetting interface.", "address", iface.PciAddress)
-					if iface.ExternallyManaged {
-						log.Log.Info("SyncNodeState(): skipping device reset as the nic is marked as externally created")
-					} else {
-						if resetErr := s.ResetSriovDevice(ifaceStatus); resetErr != nil {
-							log.Log.Error(resetErr, "SyncNodeState(): failed to reset on error SR-IOV interface")
-						}
-					}
-					return err
-				}
-
-				// Save the PF status to the host
-				err := storeManager.SaveLastPfAppliedStatus(&iface)
+				skip, err := checkNeedUpdate(&iface, &ifaceStatus, storeManager)
 				if err != nil {
-					log.Log.Error(err, "SyncNodeState(): failed to save PF applied config to host")
-					return err
+					log.Log.Error(err, "getConfigureAndReset(): failed to check interface")
+					return nil, nil, err
 				}
-				break
+				if skip {
+					break
+				}
+				toBeConfigured = append(toBeConfigured, interfaceToConfigure{iface: iface, ifaceStatus: ifaceStatus})
 			}
 		}
+
 		if !configured && ifaceStatus.NumVfs > 0 {
-			if skip := pfsToConfig[ifaceStatus.PciAddress]; skip {
-				continue
+			if skip := pfsToConfig[ifaceStatus.PciAddress]; !skip {
+				toBeResetted = append(toBeResetted, ifaceStatus)
 			}
+		}
+	}
+	return toBeConfigured, toBeResetted, nil
+}
 
-			// load the PF info
-			pfStatus, exist, err := storeManager.LoadPfsStatus(ifaceStatus.PciAddress)
-			if err != nil {
-				log.Log.Error(err, "SyncNodeState(): failed to load info about PF status for device",
-					"address", ifaceStatus.PciAddress)
-				return err
+func (s *sriov) configSriovInterfacesInParallel(storeManager store.ManagerInterface, interfaces []interfaceToConfigure) error {
+	log.Log.V(2).Info("ConfigSriovInterfacesInParallel(): start sriov configuration")
+
+	// TODO(e0ne): store all errors in SriovNetworkNodeState
+	var result error
+	errChannel := make(chan error)
+	interfacesToConfigure := 0
+	for ifaceIndex, iface := range interfaces {
+		interfacesToConfigure += 1
+		go func(iface *interfaceToConfigure) {
+			if err := s.ConfigSriovDevice(&iface.iface, &iface.ifaceStatus); err != nil {
+				log.Log.Error(err, "ConfigSriovInterfacesInParallel(): fail to configure sriov interface. resetting interface.", "address", iface.iface.PciAddress)
+				if iface.iface.ExternallyManaged {
+					log.Log.Info("ConfigSriovInterfacesInParallel(): skipping device reset as the nic is marked as externally created")
+				} else {
+					if resetErr := s.ResetSriovDevice(iface.ifaceStatus); resetErr != nil {
+						log.Log.Error(resetErr, "ConfigSriovInterfacesInParallel(): failed to reset on error SR-IOV interface")
+					}
+				}
+				errChannel <- err
```

Comment:
nit you could send to channel once in code in outer block

`errChannel <- err`

if no error, err will be nil

Reply:
in this case we need to devicne `err` variable before `if` statement. Do we want it?

Reply:
IMO yes, if it makes sense to u.

Reply:
I agree, it will make code less complex